service/discovery/azure/storage.go

```diff
@@ -386,6 +416,46 @@ func accountName(id string) string {
 	return splitName[8]
 }
 
+// idUpToStorageAccount returns the resource ID cutting of the storage type information, e.g., '/subscriptions/XXXXXXXXXXXX-XXXX-XXXX-XXXXXXXXXXXX/resourceGroups/resourceGroupName/providers/Microsoft.Storage/storageAccounts/containerName'
+func idUpToStorageAccount(id string) string {
```

Comment:
what's the point of this function? this confuses me.

Reply:
The ID is in the format `/subscriptions/XXXXXXXXXXXX-XXXX-XXXX-XXXXXXXXXXXX/resourceGroups/resourceGroupName/providers/Microsoft.Storage/storageAccounts/containerName/blobServices/default/containers/testContainer` but we need the ID as follows `/subscriptions/XXXXXXXXXXXX-XXXX-XXXX-XXXXXXXXXXXX/resourceGroups/resourceGroupName/providers/Microsoft.Storage/storageAccounts/containerName`. 

The function cuts of everything behind "containerName".

Reply:
Hm ok in the function `handleObjectStorage` you could / or should just use `account.ID` instead of `container.ID` to get the correct one. But in the vault API it seems like you only get the container ID.

Then I would suggest to at least replace the last line with `return "/" + strings.Join(split[1:9], "/")` and rename the function to something like `accountIDFromContainerID` or something so that one knows that this returns the account resource ID from a container ID.

Reply:
You're absolutely right. To be honest, I have no idea what that leftover is. I removed it.